### PR TITLE
ebs_backup: added retry for exceeding of quota

### LINKED ISSUE #44325: backup_ebs: support for retrying when the pending snapshots exceeds the quota
## Enhancement
Currently AWS restricts the pending snapshots of each region to 100. We may reach the limit when there are more than 34 nodes of TiKV.

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve?

Issue Number: close #44325

Problem Summary:
See the issue.

### What is changed and how it works?
This PR added some utilities for retrying when the snapshot count reaches the quota.

### Check List

Tests 

- [x] Manual test (add detailed scripts or steps below)
(TBD)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/aws/ebs.go [hunk 1]
```diff
@@ -64,7 +68,7 @@ func NewEC2Session(concurrency uint, region string) (*EC2Session, error) {
 		return nil, errors.Trace(err)
 	}
 	ec2Session := ec2.New(sess)
-	return &EC2Session{ec2: ec2Session, concurrency: concurrency}, nil
+	return &EC2Session{ec2: ec2Session, concurrency: concurrency, pendCache: new(pendingSnapshotCache)}, nil
```
Comment (BornChanger) [inline br/pkg/aws/ebs.go:71]:
```suggestion
	return &EC2Session{ec2: ec2Session, concurrency: concurrency)}, nil
```

**Review by BornChanger — APPROVED** [review]
We need to verify the aws error code on real scenario before we can merge this pr.

File: br/pkg/aws/ebs.go [hunk 2]
```diff
@@ -233,6 +237,18 @@ func (e *EC2Session) CreateSnapshots(backupInfo *config.EBSBasedBRMeta) (map[str
 	return snapIDMap, volAZs, nil
 }
 
+func (e *EC2Session) createSnapshotsWithRetry(ctx context.Context, input *ec2.CreateSnapshotsInput) (*ec2.CreateSnapshotsOutput, error) {
+	res, err := e.ec2.CreateSnapshotsWithContext(ctx, input)
+	if aerr, ok := err.(awserr.Error); ok && aerr.Code() == errCodeTooManyPendingSnapshots {
+		time.Sleep(pollingPendingSnapshotInterval)
```
Comment (BornChanger) [inline br/pkg/aws/ebs.go:243]: add a log entry here.

File: br/pkg/aws/ebs.go [hunk 3]
```diff
@@ -233,6 +237,18 @@ func (e *EC2Session) CreateSnapshots(backupInfo *config.EBSBasedBRMeta) (map[str
 	return snapIDMap, volAZs, nil
 }
 
+func (e *EC2Session) createSnapshotsWithRetry(ctx context.Context, input *ec2.CreateSnapshotsInput) (*ec2.CreateSnapshotsOutput, error) {
+	res, err := e.ec2.CreateSnapshotsWithContext(ctx, input)
+	if aerr, ok := err.(awserr.Error); ok && aerr.Code() == errCodeTooManyPendingSnapshots {
```
Comment (BornChanger) [inline br/pkg/aws/ebs.go:242]: add a log entry here.

**Review by Leavrth — APPROVED** [review]

**Pull request merged**